src/main/java/com/linkedin/kmf/services/SignalFxMetricsReporterService.java

```diff
@@ -164,34 +160,35 @@ private void captureMetrics() {
         parts = parts[0].split("=");
         if (parts.length < 2 || !parts[1].contains("cluster-monitor")) {
           continue;
-        }
-        String signalFxMetricName = generateSignalFxMetricName(attributeValue.mbean(), attributeValue.attribute());        
-        LOG.info("Sending metric name : " + signalFxMetricName + " and value : " + attributeValue.value());
-        setMetricValue(signalFxMetricName, attributeValue.value());
+        }                
+        setMetricValue(attributeValue);
       }      
     }
   }
   
-  private void setMetricValue(String metric, double value) {
-    if (!_metricMap.containsKey(metric)) {
-      createMetric(metric);
-    }
-    _metricMap.get(metric).setValue(value);
+  private void setMetricValue(MbeanAttributeValue attributeValue) {
+    String key = attributeValue.mbean() + attributeValue.attribute();
+    if (!_metricMap.containsKey(key)) {
+      createMetric(attributeValue);
+    }    
+    _metricMap.get(key).setValue(attributeValue.value());
   }
   
-  private void createMetric(String metric) {
-    LOG.info("Creating metric : " + metric);
+  private void createMetric(MbeanAttributeValue attributeValue) {
+    String signalFxMetricName = generateSignalFxMetricName(attributeValue.mbean(), attributeValue.attribute());
+    LOG.info("Creating metric : " + signalFxMetricName);
     SettableDoubleGauge s = _metricMetadata.forMetric(new SettableDoubleGauge())
-        .withMetricName(metric).metric();    
+        .withMetricName(signalFxMetricName).metric();    
     for (String key : _dimensionsMap.keySet()) {
       String value = _dimensionsMap.get(key);
       _metricMetadata.forMetric(s).withDimension(key, value);
     }
-    if (metric.contains("partition")) {          
-      String partitionNumber = "" + metric.charAt(metric.length() - 1);
+    if (signalFxMetricName.contains("partition")) {          
```

Comment:
If you're extracting the partition number as a dimension, you should probably also remove it from the metric name?